source/common/router/retry_state_impl.cc

```diff
@@ -187,6 +190,12 @@ bool RetryStateImpl::wouldRetry(const Http::HeaderMap* response_headers,
     }
   }
 
+  if (retry_on_ & RetryPolicy::RETRY_ON_GATEWAY_ERROR) {
+    if (Http::CodeUtility::isGatewayError(Http::Utility::getResponseStatus(*response_headers))) {
```

Comment:
You need to check here that response_headers is not nullptr. Please add a test to cover this case.

Reply:
Hi, @mattklein123 thanks for the review. Since this policy has the same behavior with 5xx, I assume I need to have reset overflow and remote reset tests in place?

Reply:
Or vice versa? Hum, it seems so. So I think, regarding this policy, I need to not retry if the response_headers is nullptr. Is that correct?